cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -264,11 +266,76 @@ private DiskEncryptionSetInner pollDiskEncryptionSetCreation(AuthenticatedContex
         return diskEncryptionSetCreationPoller.startPolling(authenticatedContext, checkerContext, desInitial);
     }
 
-    private void grantKeyVaultAccessPolicyToDiskEncryptionSetServicePrincipal(AzureClient azureClient, String vaultResourceGroupName, String vaultName,
-            String desResourceGroupName, String desName, String desPrincipalObjectId) {
+    private void grantKeyVaultRoleToDiskEncryptionSetServicePrincipal(AzureClient azureClient, String vaultResourceGroupName,
+            String vaultName, String diskEncryptionSetPrincipalObjectId, String description, CloudContext cloudContext) {
+        try {
+            LOGGER.info("Granting {}.", description);
+            String roleId = azureClient.grantKeyVaultRoleToServicePrincipal(vaultResourceGroupName, vaultName,
+                    diskEncryptionSetPrincipalObjectId);
+            LOGGER.info("Granted {}.", description);
+            CloudResource roleCloudResource = CloudResource.builder()
```

Comment:
As mentioned in other comments, this should be named like `roleAssignmentCloudResource` or something similar along those lines.